### PR TITLE
feat: support mounting blobs during copy

### LINKED ISSUE #1238: Add mount support for `oras cp`
### What is the version of your ORAS CLI

main

### What would you like to be added?

When the source and destination registries are the same, the code needs to simply needs to tell oras-go via the oras.CopyOptions that we want to mount from the source repository if available.

An example of how to do this is [here](https://github.com/oras-project/oras-go/blob/main/example_copy_test.go#L218).  You need to implement the `MountFrom` function and have it return the source repository (if the registry is the same as the destination).  You probably also want to implement `OnMount`.

### Why is this needed for ORAS?

There is a large performance improvement possible if mounting is supported.  `crane cp` has always supported mounting and now that oras-go supports mounting (via [PR](https://github.com/oras-project/oras-go/pull/665)), we should add it to `oras cp`.

### Are you willing to submit PRs to contribute to this feature?

- [ ] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR add mount support for `oras cp`. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1238 

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: cmd/oras/internal/display/track/target.go [hunk 1]
```diff
@@ -66,6 +67,14 @@ func NewTarget(t oras.GraphTarget, actionPrompt, donePrompt string, tty *os.File
 	return gt, nil
 }
 
+// Mount makes graphTarget implement oras.Mounter.
+func (t *graphTarget) Mount(ctx context.Context, desc ocispec.Descriptor, fromRepo string, getContent func() (io.ReadCloser, error)) error {
+	if mounter, ok := t.GraphTarget.(registry.Mounter); ok {
+		return mounter.Mount(ctx, desc, fromRepo, getContent)
+	}
+	return errors.New("not implemented")
+}
```
Comment (shizhMSFT) [inline cmd/oras/internal/display/track/target.go:76]: Do we need a new structure to make OCI layout scenarios work?

Comment (qweeah) [reply cmd/oras/internal/display/track/target.go:76]: OCI layout will be blocked by `opts.MountFrom`

Comment (qweeah) [reply cmd/oras/internal/display/track/target.go:76]: Can you elaborate more about `work`? If there is an OCI layout involved in copying, we only need to block this method from being called. Currently it's via `opts.MountFrom`.

Comment (qweeah) [reply cmd/oras/internal/display/track/target.go:76]: Discussed offline with @shizhMSFT, will update to type assertion.

**Comment by qweeah** [comment]
@ktarplee Can you help review if this implementation met the expected design of https://github.com/oras-project/oras-go/pull/665

File: cmd/oras/internal/display/track/target.go [hunk 2]
```diff
@@ -66,6 +66,13 @@ func NewTarget(t oras.GraphTarget, actionPrompt, donePrompt string, tty *os.File
 	return gt, nil
 }
 
+// Mount makes graphTarget implement oras.Mounter.
```
Comment (shizhMSFT) [inline cmd/oras/internal/display/track/target.go:69]:
```suggestion
// Mount mounts a blob from a specified repository. This method is invoked only by the `*remote.Repository` target.
```

**Comment by ktarplee** [comment]
> @ktarplee Can you help review if this implementation met the expected design of [oras-project/oras-go#665](https://github.com/oras-project/oras-go/pull/665)

This looks great!  I did not see any issues.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**